iceoryx2/src/service/header/publish_subscribe.rs

```diff
@@ -56,8 +55,16 @@ impl Header {
         self.publisher_port_id
     }
 
-    /// Returns the [`Layout`] of the corresponding payload.
-    pub fn payload_type_layout(&self) -> Layout {
-        self.payload_type_layout
+    /// Returns how many elements are stored inside the sample's payload.
+    ///
+    /// # Details when using
+    /// [`CustomPayloadMarker`](crate::service::builder::publish_subscribe::CustomPayloadMarker)
+    ///
+    /// In this case the number of elements relates to the element defined in the
+    /// [`MessageTypeDetails`](crate::service::static_config::message_type_details::MessageTypeDetails).
+    /// When the element has a `payload.size == 40` and the `Sample::payload().len() == 120` it
```

Comment:
This is what I mean, the term `payload` is used in multiple contexts and a comment is necessary to point out what it means. In this case it would make more sense to talk about `element.size` or `item.size`. A payload can consist of one or multiple elements/items and all elements make up the payload size.

Reply:
I vote for "element"

Reply:
I agree, I just speak about `payload.size` since this is the member of `MessageTypeDetails`.

I will refactor it so that it becomes more clear.

Reply:
I tried refactoring this, but this will be a huge task and there are a lot of files affected. I think it would be best to have a separate session for this.

>  A payload can consist of one or multiple elements/items and all elements make up the payload size.

This is only true for the sliced API. Otherwise, the element is the payload, and it will lead to confusion when only working with this API.

Reply:
Btw. we are discussing here a comment that becomes only relevant for an internal API that the user shall never use.
I acknowledge that this is important and we have to refine the code. But making the code reflecting the terms correctly without increasing the code complexity is a challenge here in my opinion.

Reply:
The discussion is about spending some time now to reduce cognitive load for us and other maintainers later.

Although it is internal API, there is still an audience, so clarity still has its merits. In the absolute worst case, we slow down contributions or even scare away potential contributors due to the amount of inside information required.

Leave it up to you to decide if it is worth doing here.

Reply:
@elfenpiff Let's adjust this in a follow-up if required.